debugger/src/gdb_scripts/linked_list_things.py

```diff
@@ -7,20 +7,100 @@
 '''
 
 import gdb
-import json
+import subprocess
+from pycparser import parse_file, c_ast
+import re
+
+# C file storing malloc line
+MALLOC_TEST_FILE = "user_malloc.c"
+
+# File to write the preprocessed C code to, before parsing with pycparser
+MALLOC_PREPROCESSED = "malloc_preprocessed"
+
+
+def remove_non_standard_characters(input_str):
+    # Remove color codes and non-standard characters using a regular expression
+    clean_str = re.sub(r'\x1b\[[0-9;]*[mK]', '', input_str)
+    return clean_str
+
+
+# Define a custom visitor class to traverse the AST and extract malloc information
+class MallocVisitor(c_ast.NodeVisitor):
+    def __init__(self):
+        self.malloc_info = []
+        self.variable_name_2 = None  # Initialize the new variable
```

Comment:
Non critical but we should give this a better name.
That can be done later in renovation though.